docs/source/rule_generation.rst

```diff
@@ -0,0 +1,65 @@
+++++++++++++++++++++++++++++++++++++++++
+Rule Generation
+++++++++++++++++++++++++++++++++++++++++
+
+The Rule generation technique is based on the folowing idea belows:
+
+1. Sort API by API usage count
+2. Separate all API into two groups, P(20% least usage count) and S(other 80% API), by the Pareto principle (20-80 rule)
+3. Combine $P$ and $S$ into four different phases:
+    - PxP
+    - PxS 
+    - SxP 
+    - SxS
+4. Execute rule generate by phase: PxP -> PxS -> SxP -> SxS
+
+The earlier the phase, the higher the value of the rule but less time spent.
+We can generate rules in a phased manner according to different situations.
+For example, under a time constraint, we can take PxP phase rules as an overview for the target APK.
+
+CLI Usage
+------------------------
+Generate rules for apk with the following command::
+
+    $ Quark -a <sample path> --generate-rule <generated rule directory path>
+
+Generate rules and web editor with the following command::
+
+    $ Quark -a <sample path> --generate-rule <generated rule directory path> -w <report file path>
```

Comment:
I suggest making the word "apk" upper case and changing "Quark" to "quark".

Also, since we concatenate the values of `-w` and `--generate-rule` in [line 383 of cli.py](https://github.com/quark-engine/quark-engine/blob/c01acc48b00163f73af783fc16d02f55eca64ad3/quark/cli.py#L383), I suggest changing the term "report file path" to "report file name". In this way, we will not mislead the user.

Reply:
I agree, but I think "web editor file name" instead of "report file name" will be more precise.